Doc/library/types.rst

```diff
@@ -98,6 +98,10 @@ the types that arise only incidentally during processing such as the
 Typical use of these names is for :func:`isinstance` or
 :func:`issubclass` checks.
 
+.. note::
+
+   These types are not supposed to be instantiated outside of CPython internals and constructor signatures will vary between Python versions.
```

Comment:
The 'not supposed to be' part may be true for some of the types not listed, but is definitely not true for at least some of the types.
```suggestion
   If you instantiate any of these types, note that signatures may vary between Python versions.
```